File: ts-sdk/src/sdk/models/operations/getabranchschema.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



export class GetABranchSchemaPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=database" })
  database: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=name" })
  name: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=organization" })
  organization: string;
}


export class GetABranchSchemaQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=keyspace" })
  keyspace?: string;
}


export class GetABranchSchemaRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetABranchSchemaPathParams;

  @SpeakeasyMetadata()
  queryParams: GetABranchSchemaQueryParams;
}


export class GetABranchSchemaResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  getABranchSchema200ApplicationJSONObject?: Record<string, Record<string, any>>;
}

File: go-sdk/pkg/models/operations/getabranchschema.go
package operations

type GetABranchSchemaPathParams struct {
	Database     string `pathParam:"style=simple,explode=false,name=database"`
	Name         string `pathParam:"style=simple,explode=false,name=name"`
	Organization string `pathParam:"style=simple,explode=false,name=organization"`
}

type GetABranchSchemaQueryParams struct {
	Keyspace *string `queryParam:"style=form,explode=true,name=keyspace"`
}

type GetABranchSchemaRequest struct {
	PathParams  GetABranchSchemaPathParams
	QueryParams GetABranchSchemaQueryParams
}

type GetABranchSchemaResponse struct {
	ContentType                              string
	StatusCode                               int64
	GetABranchSchema200ApplicationJSONObject map[string]map[string]interface{}
}
